tests/trimming/Microsoft.Kiota.Trimming.Validation/Microsoft.Kiota.Trimming.Validation.csproj

```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>net6.0;net8.0</TargetFrameworks>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <PublishTrimmed>true</PublishTrimmed>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\..\src\abstractions\Microsoft.Kiota.Abstractions.csproj" />
+    <ProjectReference Include="..\..\..\src\authentication\azure\Microsoft.Kiota.Authentication.Azure.csproj" />
+    <ProjectReference Include="..\..\..\src\http\httpClient\Microsoft.Kiota.Http.HttpClientLibrary.csproj" />
+    <ProjectReference Include="..\..\..\src\serialization\form\Microsoft.Kiota.Serialization.Form.csproj" />
+    <ProjectReference Include="..\..\..\src\serialization\json\Microsoft.Kiota.Serialization.Json.csproj" />
+    <ProjectReference Include="..\..\..\src\serialization\multipart\Microsoft.Kiota.Serialization.Multipart.csproj" />
+    <ProjectReference Include="..\..\..\src\serialization\text\Microsoft.Kiota.Serialization.Text.csproj" />
+    <TrimmerRootAssembly Include="Microsoft.Kiota.Abstractions" />
```

Comment:
From reading the docs I understand that those instructions should be enough to get everything analyzed and that we don't need a generated client to ensure it. But I'd like your opinion on the matter.
https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming?pivots=dotnet-8-0

Reply:
> TrimmerRootAssembly ensures that every part of the library is analyzed. It tells the trimmer that this assembly is a "root". A "root" assembly means the trimmer analyzes every call in the library and traverses all code paths that originate from that assembly.

Looking at the same, and I agree with this. 
We can stick with this for now and re-assess whether we want to maintain a generated client in the test in the project if we get feedback of any gaps in this approach.